bindings/python/wscript

```diff
@@ -295,50 +304,107 @@ int main ()
-        gccxml = None
-    if not gccxml:
-        Logs.warn("gccxml missing; automatic scanning of API definitions will not be possible")
-        ns3waf._report_optional_feature(conf, "pygccxml", "Python API Scanning Support", False,
-                                     "gccxml missing")
+        castxml = None
+    if not castxml:
+        Logs.warn("castxml missing; automatic scanning of API definitions will not be possible")
+        ns3waf._report_optional_feature(conf, "castxml", "Python API Scanning Support", False,
+                                     "castxml missing")
         return
 
-    gccxml_version_line = os.popen(gccxml[0] + " --version").readline().strip()
-    m = re.match( "^GCC-XML version (\d\.\d(\.\d)?)$", gccxml_version_line)
-    gccxml_version = m.group(1)
-    gccxml_version_ok = ([int(s) for s in gccxml_version.split('.')] >= [0, 9])
-    conf.msg('Checking for gccxml version', gccxml_version)
-    if not gccxml_version_ok:
-        Logs.warn("gccxml too old, need version >= 0.9; automatic scanning of API definitions will not be possible")
-        ns3waf._report_optional_feature(conf, "pygccxml", "Python API Scanning Support", False,
-                                     "gccxml too old")
+    out = subprocess.Popen([castxml[0], '--version'], 
+                            stdout=subprocess.PIPE).communicate()[0]
+    castxml_version_line = maybe_decode(out).split('\n', 1)[0].strip()
+    ## Expecting version string such as 'castxml version 0.1-gfab9c47'
+    m = re.match( "^castxml version (\d\.\d)(-)?(\w+)?", castxml_version_line)
+    try:
+        castxml_version = m.group(1)
+        castxml_version_ok = castxml_version >= REQUIRED_CASTXML_VERSION
+    except AttributeError:
+        castxml_version = castxml_version_line
+        castxml_version_ok = False
+    conf.msg('Checking for castxml version', castxml_version)
+    if not castxml_version_ok:
+        Logs.warn("castxml version unknown or too old, need version >= %s; automatic scanning of API definitions will not be possible" % REQUIRED_CASTXML_VERSION)
+        ns3waf._report_optional_feature(conf, "castxml", "Python API Scanning Support", False,
+                                     "castxml too old")
         return
 
     ## If we reached
     conf.env['ENABLE_PYTHON_SCANNING'] = True
-    ns3waf._report_optional_feature(conf, "pygccxml", "Python API Scanning Support", True, None)
+    ns3waf._report_optional_feature(conf,"castxml", "Python API Scanning Support", True, None)
 
 # ---------------------
 
 def get_headers_map(bld):
-    headers_map = {} # header => module
-    for ns3headers in bld.all_task_gen:
-        if 'ns3header' in getattr(ns3headers, "features", []):
-            if ns3headers.module.endswith('-test'):
-                continue
-            for h in ns3headers.to_list(ns3headers.headers):
-                headers_map[os.path.basename(h)] = ns3headers.module
-    return headers_map
+    module_headers = [
```

Comment:
Is there a reason we need to enumerate headers?  It seems fragile to do it this way.

Reply:
I'm sorry for the patch code there. I was using this initially while testing. I've updated the setup code in 6ebb5b6be3b1c66c18d9afbcdb8e69a72685a12a 

We actually require to iterate the headers as we need to pass a header map to Apiscan Task for it to identify the ns-3 classes we use in DCE, and since in DCE's taskgen we don't have access to ns3's headers, we need to iterate the headers. For DCE, it gets stored in the taskgen ./lib/ns3-dce and can be loaded directly.

Sir, does this setup code look good enough ?